### ISSUE TITLE
Functional component test fails at findComponent

### ISSUE THREAD

**Issue by cyberalien** [vuejs/vue-test-utils-next#187]
Simple test with functional component:

```js
import { h } from 'vue';
import { mount } from '@vue/test-utils';

function Test(props, context) {
	return h('div', {
		...props,
		innerHTML: 'This is a test',
	});
}

describe('Mounting component', () => {
	test('with wrapper', () => {
		const Wrapper = {
			components: { Test },
			template: `<Test />`,
		};

		const wrapper = mount(Wrapper, {});
		const item = wrapper.findComponent(Test);
		expect(item.exists()).toBe(true);
	});

	test('without wrapper', () => {
		const wrapper = mount(Test, {});
		const item = wrapper.findComponent(Test);
		expect(item.exists()).toBe(true);
		expect(item.html()).toStrictEqual('<div>This is a test</div>');
	});
});
```
Both tests fail at `findComponent()` that returns null.

vue: 3.0.0-rc.7
vue/test-utils: 2.0.0-beta.2, also tried with latest build from this repo

**Comment by lmiller1990** [comment]
After merging this https://github.com/vuejs/vue-test-utils-next/pull/188/ I will investigate this a little. I learned a lot in #184.

I believe this was not working in VTU v1 either. I expect some limitations to by applied to a found functional component (currently we cannot detect emitted events from them, for example).

In the meantime, I will be adding some warnings to let people know this is a known issue, such as here: https://github.com/vuejs/vue-test-utils-next/pull/185

**Comment by cyberalien** [comment]
Thanks!

As for VTU v1, it is working with similar functional component for Vue 2.

**Comment by lmiller1990** [comment]
Fixed in #238 

This will be in the next release
